vdev/src/commands/release/push.rs

```diff
@@ -0,0 +1,34 @@
+use std::env;
+
+use anyhow::Result;
+use clap::Args;
+
+use crate::util;
+use crate::git;
+
+/// Pushes new versions produced by `make release` to the repository
+#[derive(Args, Debug)]
+#[command()]
+pub struct Cli {}
+
+impl Cli {
+    pub fn exec(self) -> Result<()> {
+        let version = env::var("VECTOR_VERSION").or_else(|_| util::read_version())?;
+        let version_minor = version.split('.')
+            .take(2)
+            .collect::<Vec<&str>>()
+            .join(".");
+
+        let current_branch = git::current_branch()?;
+        println!("Preparing the branch and the tag...");
+        git::checkout_branch(&format!("v{version_minor}"))?;
```

Comment:
This is a little different than the original, in that this will fail if the branch already exists.

I think it would be handy to capture the two `format!(…)` values up front instead of generating them each time. i.e.
```rust
let foobar = format!("v{version_minor}");
git::checkout_branch(&foobar)?;
```

Reply:
I am curious as to why capture_output doesn't contain the string "fatal: blah blah" if the branch would exist already though. 

I'm testing this through vscode and seeing the values live as the program executes and stepping through. Perhaps since I have output declared as `_output` it defaults the string to "" and never changes it? I will verify this once more and get back.

Reply:
It seems like there is one behavior we can predict:

if the branch already existed, then we don't switch the branch. (silent fatal occured)
if the branch didn't exist, then we create it and switch the branch.

I added print statements before and after to confirm the behavior and stepped through the code.

```
pub fn checkout_branch(branch_name: &str) -> Result<()> {
    let prev_branch = current_branch()?;
    println!("before capture_output(checkout -b) {prev_branch}");
    let _output = capture_output(&["checkout", "-b", branch_name])?;
    let curr_branch = current_branch()?;
    println!("after capture_output(checkout -b) {curr_branch}");
    Ok(())
}

```

Reply:
I created a helper function branch_exists to implement the logic of the original script